### PR TITLE
Add configurable Middleware to query server

### PR DESCRIPTION
**Pull request by ryanhall07**
This allows configuring the Middleware that runs in the query server. By
default if the Middleware is not set, a set of default Middleware
functions are run (i.e the existing middleware)

**What this PR does / why we need it**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note

```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note

```

### CODE REVIEW

File: src/query/api/v1/options/middleware.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+		},
+	}
+}
+
+// CompressionMiddleware adds suitable response compression based on the client's Accept-Encoding headers.
+var CompressionMiddleware = func(base http.Handler) http.Handler {
+	return httputil.CompressionHandler{
+		Handler: base,
+	}
+}
```
Comment (robskillington) [inline src/query/api/v1/options/middleware.go:49]: nit: These could all be constructors which would make them immutable? (i.e. couldn't modify them from another package by doing `options.DefaultMiddleware = nil`)

i.e.
```go
func DefaultMiddlewareIO () []mux.MiddlewareFunc { return ... } 

func TracingMiddleware(tracer opentracing.Tracer) mux.MiddlewareFunc { return ... }

func CorsMiddleware() mux.MiddlewareFunc { return ... }

func CompressionMiddleware() mux.MiddlewareFunc { return ... }
```

**Review by robskillington — APPROVED** [review]
LGTM other than minor nit that can either take or leave it

File: src/query/api/v1/options/middleware.go (hunk 1 above)
Comment (ryanhall07) [reply src/query/api/v1/options/middleware.go:49]: good call!

**Review by wesleyk — APPROVED** [review]
LGTM

**Pull request merged**